torchaudio/prototype/ctc_decoder/__init__.py

```diff
@@ -1,19 +1,43 @@
-import torchaudio
-
-try:
-    torchaudio._extension._load_lib("libtorchaudio_decoder")
-    from .ctc_decoder import Hypothesis, CTCDecoder, ctc_decoder, lexicon_decoder, download_pretrained_files
-except ImportError as err:
-    raise ImportError(
-        "flashlight decoder bindings are required to use this functionality. "
-        "Please set BUILD_CTC_DECODER=1 when building from source."
-    ) from err
-
-
-__all__ = [
+_INITIALIZED = False
+_LAZILY_IMPORTED = [
     "Hypothesis",
     "CTCDecoder",
     "ctc_decoder",
     "lexicon_decoder",
     "download_pretrained_files",
 ]
+
+
+def _init_extension():
+    import torchaudio
+
+    try:
+        torchaudio._extension._load_lib("libtorchaudio_decoder")
+    except ImportError as err:
+        raise ImportError(
+            "flashlight decoder bindings are required to use this functionality. "
+            "Please set BUILD_CTC_DECODER=1 when building from source."
+        ) from err
+
+    global _INITIALIZED
+    _INITIALIZED = True
+
+
+def __getattr__(name: str):
+    if name in _LAZILY_IMPORTED:
+        if not _INITIALIZED:
+            _init_extension()
+
+        from . import _ctc_decoder
```

Comment:
@carolineechen I am renaming `ctc_decoder.py` to `_ctc_decoder.py` as it caused infinite loop when building documentation. (I suspect that having the same name for module and script cause this.)